server/src/main/java/org/opensearch/action/search/SearchRequestOperationsListener.java

```diff
@@ -10,23 +10,23 @@
 
 import org.apache.logging.log4j.Logger;
 import org.apache.logging.log4j.message.ParameterizedMessage;
-import org.opensearch.common.annotation.PublicApi;
+import org.opensearch.common.annotation.InternalApi;
 
 import java.util.List;
 
 /**
  * A listener for search, fetch and context events at the coordinator node level
  *
- * @opensearch.api
+ * @opensearch.internal
  */
-@PublicApi(since = "1.0.0")
-public interface SearchRequestOperationsListener {
+@InternalApi
+interface SearchRequestOperationsListener {
 
-    void onPhaseStart(SearchPhaseContext context);
+    void onPhaseStart(SearchPhase phase);
```

Comment:
@andrross find out that `SearchRequestOperationsListener`  (that was introduce recently) was bringing the whole `SearchPhaseContext` with it but it does not need it, it needs only `SearchPhase`, changing the API and reducing the visibility  of other APIs to `InternalApi`

Reply:
I like this change, but just wanted to call out that we may want to make `SearchRequestOperationsListener` public in future, depending on the approach used for https://github.com/opensearch-project/OpenSearch/issues/11186. Of course, we should cross that bridge if/when we reach it and keep it internal until then.

I've been chatting with @ansjcy about the approach to use there. His proposal suggests "listening" for slow/expensive queries using `SearchRequestOperationsListener`, whereas I've advocated for propagating the needed data into a `SearchResponseProcessor` and tracking there (possibly via a new field on `PipelineProcessingContext`). 

Either way, I'd like to try to keep the collection of queries outside of core (or at least outside of `/server`), in a plugin or module. The amount of extra logic we're cramming into `TransportSearchAction` and friends worries me (and my hands are certainly not clean there either).

Reply:
Thanks @msfroh , it will be possible to make it public for sure when time comes :-) For now, we could keep it package private

Reply:
The code changes faster than we are able to clean it up, reverting some API changes since there new hooks added and the assumptions do not apply anymore.